src/%IPM/ExtensionBase/Composite.cls

```diff
@@ -0,0 +1,339 @@
+/// This class provides the API and interface to work with multiple Studio extensions
+/// It directs the action and lifecycle calls to corresponding primary/secondary extension class
+/// Set Inheritance to right, so that on conflict definitions, CompositeMethodOverrides.cls will take over %Studio.Extension.Base.cls
+/// This way, CompositeMethodOverrides.cls can override final methods in %Studio.* methods
+Class %IPM.ExtensionBase.Composite Extends (%Studio.Extension.Base, %IPM.ExtensionBase.CompositeMethodOverrides) [ Inheritance = right, Owner = {%Developer} ]
```

Comment:
Why this is part of IPM?

Reply:
Good question! This is the abstract super class for %IPM.Extension.ServerMenu.MenuItems to be used during init, in order to avoid invalid references. Itself is just an abstract class but our studio ui extension, or user defined ui extension, can extend this class. So we are not adding Studio support, but just provide users with super class to add their studio support.

Reply:
I still see no reasons, of having it as part of IPM
it can be just an extra module, which then can be used by extensions developers as a dependency

Reply:
User story:
As a developer who is currently using a source control extension and *not* the package manager (yet), I want to be able to install the package manager and have it "just work" (including the Package Manager's own extension) without it breaking my development environment.

Reply:
Looks like, there is an issue with IPM then, why it's even possible to break it. Why it should touch user's SourceControl class at all? I still don't get, the reasons of having own Studio's support at all

Reply:
Server side source control hooks need to map files to different modules instead of to a single root ^Sources value which is why we have custom source control classes for working with IPM